src/test/java/ch/jalu/configme/configurationdata/CommentsConfigurationTest.java

```diff
@@ -11,12 +11,31 @@
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 /**
  * Test for {@link CommentsConfiguration}.
  */
 class CommentsConfigurationTest {
 
+    @Test
+    void shouldThrowForExistingPath() {
+        // given
+        final String path = "config.me"; 
+        CommentsConfiguration conf = new CommentsConfiguration();
+        conf.setComment(path, "Old", "Comments", "1", "2", "3");
+        
+        // when
+        IllegalStateException ex = assertThrows(IllegalStateException.class, () -> conf.setComment(path, "New Comment"));
+
+        // then
+        assertThat(ex.getMessage(), equalTo("Comment lines already exists for the path " + path));
+        assertEquals(conf.getAllComments().size(), 1);
+        assertEquals(conf.getAllComments().get(path).size(), 1);
```

Comment:
I try to use Hamcrest matchers everywhere so that all assertions can be read as `assertThat(testee, matcherWithExpectation())`. There are some exceptions with `assertTrue` and `assertFalse` when we need to do a lot of checks on booleans and nothing else (i.e. they're not mixed with `assertThat`)

Here I would suggest to maybe change L34 to assert that the key set has the entry we expect:
```
assertThat(conf.getAllComments().keySet(), contains(path));
```

^ That's an implicit check for the size, because `contains` will fail if there's any entry it can't account for, and so by that logic L35 would not be needed (L36 has an implicit size check).